File: projects/order-creation-queries/core/src/main/java/com/robinfood/core/mappers/OriginMappers.java
package com.robinfood.core.mappers;

import com.robinfood.core.dtos.OriginDTO;
import com.robinfood.core.entities.services.OriginEntity;

public final class OriginMappers {

    private OriginMappers() {
        throw new IllegalStateException("Utility class");
    }

    public static OriginDTO originEntityToOriginDto(OriginEntity originEntity) {

        return  OriginDTO.builder()
                .color(originEntity.getColor())
                .id(originEntity.getId())
                .name(originEntity.getName())
                .build();
    }
}

File: projects/posv2/core/src/main/kotlin/com/robinfood/core/mappers/OriginMappers.kt
package com.robinfood.core.mappers

import com.robinfood.core.dtos.transactionrequest.OriginDTO
import com.robinfood.core.entities.transactionrequest.OriginEntity

fun OriginDTO.toOriginEntity(): OriginEntity {
    return OriginEntity(
        id = id,
        name = name
    )
}

fun OriginEntity.toOriginDTO(): OriginDTO {
    return OriginDTO(
            id = id,
            name = name
    )
}